### PR TITLE
Bp feature/add user id to ga 2

### PR DESCRIPTION
**Pull request by BarryPiccinni**

### CODE REVIEW

File: src/modules/ga/index.js [hunk 1]
```diff
@@ -15,7 +15,9 @@ function createCicaGa(window) {
         action: 'click', // <String>
         category: 'category', // <String>
         label: undefined, // <String>
-        value: undefined // non-negative <Integer>
+        value: undefined, // non-negative <Integer>
+        timestamp: undefined,
```
Comment (armoj) [inline src/modules/ga/index.js:19]: BP comment: not needed

File: src/modules/ga/index.js [hunk 2]
```diff
@@ -31,7 +33,9 @@ function createCicaGa(window) {
             event_label: gtagOptions.label,
             value: gtagOptions.value,
             event_callback: gtagOptions.callback,
-            non_interaction: gtagOptions.nonInteraction
+            non_interaction: gtagOptions.nonInteraction,
+            transaction_id: gtagOptions.transaction_id,
+            event_timestamp: gtagOptions.timestamp
```
Comment (armoj) [inline src/modules/ga/index.js:38]: BP comment: not needed

File: src/modules/ga/index.js [hunk 3]
```diff
@@ -117,6 +121,54 @@ function createCicaGa(window) {
...
+                'submit',
+                () => {
+                    const gtagOptions = {
+                        type: 'event',
+                        action: 'conversion',
+                        category: 'application_journey_tracker',
+                        label: 'finish',
+                        nonInteraction: true,
+                        transaction_id: window.CICA.CICA_ANALYTICS_ID,
+                        timestamp: Date.now()
```
Comment (armoj) [inline src/modules/ga/index.js:163]: BP comment: not needed

File: src/modules/ga/index.js [hunk 4]
```diff
@@ -117,6 +121,54 @@ function createCicaGa(window) {
...
+            send(gtagOptions);
+        } else if (window.location.pathname.includes('sign-in/success')) {
+            const gtagOptions = {
+                type: 'event',
+                action: 'conversion',
+                category: 'application_journey_tracker',
+                label: 'save',
+                nonInteraction: true,
+                transaction_id: window.CICA.CICA_ANALYTICS_ID,
+                timestamp: Date.now()
```
Comment (armoj) [inline src/modules/ga/index.js:149]: BP comment: not needed

File: src/modules/ga/index.js [hunk 5]
```diff
@@ -117,6 +121,54 @@ function createCicaGa(window) {
...
+        const isFirstPage = new URLSearchParams(window.location.search).get('utm_source');
+        if (isFirstPage) {
+            const gtagOptions = {
+                type: 'event',
+                action: 'conversion',
+                category: 'application_journey_tracker',
+                label: 'start',
+                nonInteraction: true,
+                transaction_id: window.CICA.CICA_ANALYTICS_ID,
+                timestamp: Date.now()
```
Comment (armoj) [inline src/modules/ga/index.js:138]: BP comment: not needed

File: src/modules/ga/index.js [hunk 6]
```diff
@@ -117,6 +121,54 @@ function createCicaGa(window) {
...
+            send(gtagOptions);
+        } else if (window.location.pathname.includes('declaration')) {
+            window.addEventListener(
+                'submit',
+                () => {
+                    const gtagOptions = {
+                        type: 'event',
+                        action: 'conversion',
+                        category: 'application_journey_tracker',
+                        label: 'finish',
```
Comment (armoj) [inline src/modules/ga/index.js:160]: should we call this 'complete'

File: test/test-fixtures/transformations/resolved html/p-applicant-british-citizen-or-eu-national.js [hunk 7]
```diff
@@ -52,7 +52,10 @@ const html = `<!DOCTYPE html>
         }
         gtag('js', new Date());
         gtag('set', {cookie_flags: 'SameSite=Lax;Secure'});
-        gtag('config', '');
+        gtag('config', '', {
+          'user_id': '',
+          'debug_mode': true
```
Comment (armoj) [inline test/test-fixtures/transformations/resolved html/p-applicant-british-citizen-or-eu-national.js:57]: remove debug mode

File: questionnaire/form-helper.js [hunk 8]
```diff
@@ -33,11 +33,14 @@ function renderSection({
...
     isAuthenticated,
-    showSignInLink = shouldShowSignInLink(sectionId, uiSchema, isAuthenticated)
+    showSignInLink = shouldShowSignInLink(sectionId, uiSchema, isAuthenticated),
+    userId,
+    analyticsId
 }) {
     const showButton = !isFinal;
     const buttonTitle = getButtonText(sectionId);
     const hasErrors = transformation.hasErrors === true;
+    const userUrn = isAuthenticated ? userId : undefined;
```
[the remark this answers is not in the document]
Comment (BarryPiccinni) [reply questionnaire/form-helper.js]: This logic was actually still needed but I've moved it to the nunjucks options object to avoid creating this `const`

**Review by armoj — APPROVED** [review]

**Pull request merged**